samlauthenticator/samlauthenticator.py

```diff
@@ -626,18 +627,48 @@ def _check_username_and_add_user(self, username):
                 self.check_blacklist(username) and \
                 self.check_whitelist(username):
             if self.create_system_users:
-                return self._optional_user_add(username)
+                if self._optional_user_add(username):
+                    # Successfully added user
+                    return username
+                else:
+                    # Failed to add user
+                    self.log.error('Failed to add user by calling add user')
+                    return None
 
-            return True
+            # Didn't try to add user
+            return username
 
-        return False
+        # Failed to validate username or failed list check
+        self.log.error('Failed to validate username or failed list check')
+        return None
```

Comment:
Instead of having this function return a boolean, I think it's more literate to have the function return `str` or `None`, and return this function (new line 696).